components/admin_console/request_button/request_button.tsx

```diff
@@ -16,122 +14,130 @@ import WarningIcon from 'components/widgets/icons/fa_warning_icon';
  * its outcome as either success, or failure accompanied by the
  * `message` property of the `err` object.
  */
-export default class RequestButton extends React.PureComponent {
-    static propTypes = {
+type Props = {
+
+    /**
+     * ID to assign to the form
+     */
+    id?: string;
+
+    /**
+     * The action to be called to carry out the request.
+     */
+    requestAction: (
+        success: (data?: any) => void,
+        error: (error: any) => void
+    ) => void;
+
+    /**
+     * A component that displays help text for the request button.
+     *
+     * Typically, this will be a <FormattedMessage/>.
+     */
+    helpText?: React.ReactNode;
+
+    /**
+     * A component to be displayed on the button.
+     *
+     * Typically, this will be a <FormattedMessage/>
+     */
+    loadingText?: React.ReactNode;
```

Comment:
this prop was originally defined as a string in `jsx` version but I guess that was a mistake? so I changed that to `ReactNode` according to the description comment.